### PR TITLE
Improving Code-execution notebook

### PR DESCRIPTION
**Pull request by Giom-V**
* Adding more explanation texts
* Adding a "next steps" section at the end

### CODE REVIEW

File: quickstarts/Code_Execution.ipynb [hunk 1]
```diff
@@ -818,6 +831,15 @@
         "chat = model.start_chat()"
       ]
     },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "9b2d47bf0927"
+      },
+      "source": [
+        "This time, you're going to ask the model to use a [Bogo-sort](https://en.wikipedia.org/wiki/Bogosort) algorythm to sort a list of numbers."
```
Comment (markmcd) [inline quickstarts/Code_Execution.ipynb:840]:
```suggestion
        "This time, you're going to ask the model to use a [Bogo-sort](https://en.wikipedia.org/wiki/Bogosort) algorithm to sort a list of numbers."
```

nit: typo

File: quickstarts/Code_Execution.ipynb [hunk 2]
```diff
@@ -994,6 +1016,15 @@
         "Markdown(response.text)"
       ]
     },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "69fc4457ffd4"
+      },
+      "source": [
+        "Try running the previous cell multiple times and you'll see a different number of iteration, indicating that Gemini API indeed ran the code and optained different results due to the nature of the algorithm."
```
Comment (markmcd) [inline quickstarts/Code_Execution.ipynb:1025]:
```suggestion
        "Try running the previous cell multiple times and you'll see a different number of iterations, indicating that the Gemini API indeed ran the code and obtained different results due to the nature of the algorithm."
```

tiny grammar nits

File: quickstarts/Code_Execution.ipynb [hunk 3]
```diff
@@ -1498,6 +1529,22 @@
...
+      },
+      "source": [
+        "## Next Steps\n",
+        "### Useful API references:\n",
+        "\n",
+        "Check the [Code execution documentation](https://ai.google.dev/gemini-api/docs/code-execution) for more details about the feature and in particular, the [recommendations](https://ai.google.dev/gemini-api/docs/code-execution?lang=python#code-execution-vs-function-calling) regarding when to use it instead of [function calling](https://ai.google.dev/gemini-api/docs/function-calling).\n",
+        "\n",
+        "### Continue your discovery of the Gemini API\n",
+        "\n",
+        "Learn how to control how the Gemini API can call your own functions using the [function calling](https://github.com/google-gemini/cookbook/blob/main/quickstarts/Function_calling.ipynb) feature, or discover how to control the model output in [JSON](https://github.com/google-gemini/cookbook/blob/main/quickstarts/JSON_mode.ipynb) or using an [Enum](https://github.com/google-gemini/cookbook/blob/main/quickstarts/Enum.*ipynb*)."
```
Comment (markmcd) [inline quickstarts/Code_Execution.ipynb:1546]: The enum link is broken.

Comment (markmcd) [inline quickstarts/Code_Execution.ipynb:1546]: Can you use relative links when referencing content in GitHub?

Doing so means that the links work:
* Across forks, branches, specific-commit views, etc
* In Colab
* In GitHub Codespaces

**Review by markmcd — DISMISSED** [review]
Just a couple of nits

File: quickstarts/Code_Execution.ipynb (hunk 3 above)
Comment (Giom-V) [reply quickstarts/Code_Execution.ipynb:1546]: I thought we wanted the opposite but sure, that a good reason.

**Review by markmcd — APPROVED** [review]

**Pull request merged**